src/bindings/c/include/openvino/c/openvino.h

```diff
@@ -15,6 +15,12 @@
  **/
 #pragma once
 
+#ifdef _WINDOWS_
+#pragma message("The BOOLEAN define in ov_element_type_e conflict with Windows.h BOOLEAN define. Here redifine the BOOLEAN of Windows.h to WIN_BOOLEAN & the BOOLEAN of ov_element_type_e to OV_BOOLEAN.")
+typedef BOOLEAN WIN_BOOLEAN;
+#define BOOLEAN OV_BOOLEAN
+#endif
```

Comment:
can we use it in some test?

Reply:
It is compilation conflict error? Is there any similar test cases for reference?

Reply:
I supposed we can emulate situation when Windows .h is used and we used our WA suggestions (OV_BOOLEAN) and it worked.

BTW, is such code cross-platform? E.g. I used OV_BOOLNEAN on Windows, what will it compile on Linux?

In general, I want to use usage scenario of our WA.

Reply:
OV_BOOLNEAN is only for Windows, it will not compile on Linux, BOOLEAN for Linux.

Reply:
Maybe we can do this define "#define BOOLEAN OV_BOOLEAN" for Windows & Linux together, which means:

**- scenario 1: On windows OS, without include windows.h, the following statements work fine (BOOLEAN = OV_BOOLEAN)**
ov_element_type_e element_type = OV_BOOLEAN;
ov_element_type_e element_type = BOOLEAN;

**- scenario 2: On windows OS, with include windows.h, the following statements work**
ov_element_type_e element_type = OV_BOOLEAN;
ov_element_type_e element_type = BOOLEAN;
WIN_BOOLEAN win_boolean;

**- scenario 3: On Linux OS, the following statements work fine (BOOLEAN = OV_BOOLEAN)**
ov_element_type_e element_type = OV_BOOLEAN;
ov_element_type_e element_type = BOOLEAN;

@ilya-lavrenov  how do u think?

Reply:
I agree.
But I still think we don't need to define `WIN_BOOLEAN`

Reply:
Ok! Let me update

Reply:
@zhaixuejun1993  One option is to add it to message that user can "typedef BOOLEAN WIN_BOOLEAN;" here if user need to use Windows BOOLEAN.

Reply:
Agree to not define `WIN_BOOLEAN` for Windows. Let's clarify the message that Window BOOLEAN cannot be used in this file. If the user does need to use Window BOOLEAN, we expect that the user should use it in a separate file.